Functions/Gherkin.Core.ps1

```diff
@@ -52,6 +52,86 @@ function Invoke-GherkinHook {
     }
 }
 
+function New-GherkinPesterState {
+    [CmdletBinding()]
+    param (
+        [string[]]$ScenarioName = [string[]]@(),
+        [string[]]$Tag = [string[]]@(),
+        [string[]]$ExcludeTag = [string[]]@(),
+        [System.Management.Automation.SessionState]$SessionState = $null,
+        [switch]$Strict,
+        [Pester.OutputTypes]$Show = 'All',
+        [hashtable]$PesterOption = $null
+    )
+
+    $AddMember = $SafeCommands['Add-Member']
+    $NewObject = $SafeCommands['New-Object']
+
+    New-PesterState -TagFilter $Tag -ExcludeTagFilter $ExcludeTag -TestNameFilter $ScenarioName -SessionState $SessionState -Strict:$Strict  -Show $Show -PesterOption $PesterOption |
+        & $AddMember -MemberType NoteProperty -Name Features -Value (& $NewObject System.Collections.Generic.List[PSObject] ) -PassThru |
+        & $AddMember -MemberType ScriptProperty -Name FailedScenarios -PassThru -Value {
+            [string[]]@(
+                $this.TestResult |
+                    & $SafeCommands['Group-Object'] Context |
```

Comment:
Interesting, so it seems `Group-Object` in PowerShell Core (6.1) ends up _sorting_ the groups by the property(ies) selected for grouping! This causes the unit tests to fail on the AppVeyor build. So I'm left wondering what my options are.

I could just force the issue and sort the results. But I'm concerned that doing this would have unintended consequences?? Is the order important in the result object? For example, the ordering on PSv2 - PSv5 is the order in which the tests were executed. Is it OK to change this ordering?

I could "synthesize" a name to preserve the ordering when using `Group-Object` on PSv6. The downside to this is the extra processing involved.

Reply:
I think I've decided I'm going to synthesize a key name upon which to perform grouping which would preserve the ordering when using `Group-Object` in PowerShell Core. I guess I should also file an issue with the PowerShell team.

Reply:
OK, so apparently, this is a breaking change in PowerShell v6. In order to dramatically improve performance from quadratic to `n log n`, they first sort the objects and then group. I did post a comment on the issue ([#7409](https://github.com/PowerShell/PowerShell/issues/7409?ts=2)), but I doubt that'll be reversed since 1) it results in dramatically better performance and 2) the work was already done, discussed, approved, and obviously released. So my decision to synthesize a key to guarantee the ordering is the right one.

On a side note, the extra processing should be minimal. The biggest cost is looping over all the results and the creation of objects in order to keep the original ordering. It's just a shame that this extra work needs to be done at all, as it clutters up the code.

Reply:
@fourpastmidnight I am not that familiar with the Gherkin code, but is preserving the ordering important? Is preserving the ordering a guarantee of the Pester implementation? 

@nohwnd @Jaykul Any input to add?

Reply:
@renehernandez I have nothing.

Reply:
I may have managed to find my way around this, and with a _MUCH_ simpler solution. I've noticed for some time now, that even though we return (basically) a Tuple  from `Import-GherkinFeature`, that the actual `$Feature` object itself has the scenarios added as a property.

With all my changes that I'm making, I'm adding a synthesized property for the scenario result as a whole. Now, as long as PSv2 doesn't drop that at some point before we get to (or as I'm) creating the TestResults, then I can simply use that to find scenarios based on their result, instead of all this hoopla. And the ordering will remain in tact, so the question becomes moot. The real question here is will this work with PSv2.

Reply:
_RIGHT_, since I added the synthetic Result property, it was subsequently dropped by PSv2. And so in `Invoke-GherkinStep`, I had to check for the property, and if it didn't exist, add it back. BUT, when I do that, of course, I never "replace" the Scenario object on the Feature with the "new" PSObject created by adding the synthetic property--because I don't have access to the feature. IOW, this won't work. Too bad. In v5, this'll totally work, though (although, things will be much different in v5 if I can ever get time to work on it).

Reply:
Hmm, I think we can work around the PSv2 issue by adding `-PassThru` to the `Add-Member` invocation in `Convert-Tags` and subsequently returning the "new" PSObject created by that command. Then I may be able to get rid of the check in `Invoke-GherkinStep`, as well as make this commit work, too.

Reply:
OK, So, I managed to work around everything by making sure that Pester doesn't lose the `PSObject` wrapper over `Gherkin.Ast.Scenario` and `Gherkin.Ast.Background` objects for extended properties that are added (e.g. the `Result` property). Doing this allows me to get rid of the check for this property in `Invoke-GherkinStep`. This also means that this property is available on the `Scenario` objects in the Pester Results object, which allows me to use _MUCH_ simpler code that works for PSv2 all the way through PSv6 without any shenanigans.

The most recent Azure Pipelines build failed because one of the steps in the PSv2 build (the finalize step) took too long to run for some reason--but all the tests passed. So now I just need to clean up the most recent 4 commits and squash them.

Reply:
This is resolved with the latest commit.